src/BenchmarkDb/Program.cs

```diff
@@ -0,0 +1,262 @@
+
+                    Interlocked.Exchange(ref _stopping, 1);
+                    stopTime = DateTime.UtcNow;
+                })
+            );
+
+            Task.WhenAll(tasks).GetAwaiter().GetResult();
+
+            var totalTps = (int) (totalTransactions / (stopTime - startTime).TotalSeconds);
+
+
+            listOfResults.Remove(listOfResults.Max());
+            listOfResults.Remove(listOfResults.Min());
+            var stddev = CalculateStdDev(listOfResults, listOfResults.Count);
+
+            Console.SetCursorPosition(0, Console.CursorTop);
+            Console.WriteLine($"{numThreads:D2} Threads, tps: {totalTps:F2}, stddev(w/o best+worst): {stddev:F2}");
+
+            var desc = $"{database}+{mode}+{numThreads}";
+
+            using (var sw = File.AppendText("results.md"))
+            {
+                sw.WriteLine($"|{desc}|{mode}|{numThreads:D2}|{totalTps:F0}|{stddev:F0}|");
+            }
+
+            using (var sw = File.AppendText("results.csv"))
+            {
+                sw.WriteLine($"{desc},{mode},{numThreads:D2},{totalTps:F0},{stddev:F0}");
+            }
+
+            double CalculateStdDev(IEnumerable<double> values, int count)
+            {
+                var avg = values.Sum() / count;
+                double sum = values.Sum(d => Math.Pow(d - avg, 2));
+                return Math.Sqrt((sum) / count);
+            }
+
+            async Task DoWorkAsync()
+            {
+                while (_stopping != 1)
+                {
+                    Interlocked.Increment(ref _counter);
+
+                    try
+                    {
+                        var results = new List<Fortune>();
+                        using (var connection = factory.CreateConnection())
+                        {
+                            connection.ConnectionString = connectionString;
+                            await connection.OpenAsync();
+
+                            using (var command = connection.CreateCommand())
+                            {
+                                command.CommandText = SqlQuery;
+                                command.Prepare();
+
+                                using (var reader = await command.ExecuteReaderAsync())
+                                {
+                                    while (await reader.ReadAsync())
+                                    {
+                                        results.Add(new Fortune
+                                        {
+                                            Id = reader.GetInt32(0),
+                                            Message = reader.GetString(1)
+                                        });
+                                    }
+                                }
+                            }
+                        }
+
+                        if (results.Count() != 12)
+                        {
+                            throw new ApplicationException("Not 12");
```

Comment:
Perhaps this is just a Debug.Assert?

Reply:
Seems to me that keeping the check in non-debug (i.e. most) runs could be useful, assuming we are likely to break things and have bad runs (e.g. db not initialized). Once we switch to Count property this should be a pretty cheap check to leave in place, correct?

Reply:
We can keep it, but we should probably switch the exception type - ApplicationException is generally discouraged.